### PR TITLE
All external mission loading put inside mission loader

### PR DESCRIPTION
**Pull request by Christdej**

### CODE REVIEW

File: frontend/src/api/ApiCaller.tsx [hunk 1]
```diff
@@ -43,6 +41,7 @@ export class BackendAPICaller {
             method,
             headers,
             mode: 'cors',
+            // credentials: 'include', // TODO CHECK IF CORRECT
```
[the remark this answers is not in the document]
Comment (Christdej) [reply frontend/src/api/ApiCaller.tsx]: Done =)

File: backend/api/Configurations/CustomServiceConfigurations.cs [hunk 2]
```diff
@@ -124,5 +125,33 @@ IConfiguration configuration
...
             return services;
         }
+
+        public static IServiceCollection ConfigureMissionLoader(
+            this IServiceCollection services,
+            IConfiguration configuration
+        )
+        {
+            string? missionLoaderFileName = configuration["MissionLoader:FileName"];
+            Console.WriteLine(missionLoaderFileName);
```
Comment (andchiind) [inline backend/api/Configurations/CustomServiceConfigurations.cs:135]: This should either be removed or logged with a proper logger

File: backend/api/Services/MissionLoaders/EchoMissionResponse.cs [hunk 3]
```diff
@@ -81,7 +81,7 @@ public class SensorType
     public class EchoPose
     {
         [JsonPropertyName("poseId")]
-        public int PoseId { get; set; }
+        public int? PoseId { get; set; }
```
Comment (andchiind) [inline backend/api/Services/MissionLoaders/EchoMissionResponse.cs:84]: Why is this nullable?

File: backend/api/Services/MissionLoaders/MissionLoaderInterface.cs [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+{
+    public interface IMissionLoader
+    {
+        public Task<MissionDefinition?> GetMissionById(string sourceMissionId);
+
+        public Task<IQueryable<MissionDefinition>> GetAvailableMissions(string? installationCode);
+
+        public Task<List<MissionTask>> GetTasksForMission(string sourceMissionId);
+
+        public Task<List<PlantInfo>> GetPlantInfos(); // Facility service
```
Comment (andchiind) [inline backend/api/Services/MissionLoaders/MissionLoaderInterface.cs:13]: I am not sure if this comment is very helpful right now

File: frontend/best_practises.md [hunk 5]
```diff
@@ -1,50 +1,51 @@
 # Flotilla frontend development best practises
 
-- [Flotilla frontend development best practises](#flotilla-frontend-development-best-practises)
```
Comment (andchiind) [inline frontend/best_practises.md:3]: A lot of these changes to READMEs and whatnot are very nice, but would it not be good to seperate them into different commits?

File: frontend/best_practises.md [hunk 6]
```diff
@@ -80,8 +82,9 @@ export const MissionTitleComponent = ({ missionName } : MissionTitleComponentPro
 ```
 
 Bad:
+
 ```
-export const RobotComponent = ({ mission, setColor } : {mission: CondensedMissionDefinition, setColor: (c: string) => void}) => {
```
Comment (andchiind) [inline frontend/best_practises.md:84]: Though I understand that this is needed to be in the same commit as the other frontend changes

File: frontend/best_practises.md (hunk 5 above)
Comment (andchiind) [reply frontend/best_practises.md:3]: Only a suggestion, not a hard requirement. Just to generally split up the big commit somewhat

File: backend/api.test/Database/Models.cs [hunk 7]
```diff
@@ -71,62 +71,5 @@ public void TestNegativaRotation()
                 3.0
             );
         }
-
-        [Fact]
-        public void AssertCoordinateConversion()
```
Comment (andchiind) [inline backend/api.test/Database/Models.cs:76]: Have these tests been moved somewhere or just deleted? Sorry it was hard to tell

**Review by andchiind — COMMENTED** [review]
Looks good otherwise. Seems to work well when testing it

File: backend/api/Services/MissionLoaders/EchoMissionResponse.cs (hunk 3 above)
Comment (Christdej) [reply backend/api/Services/MissionLoaders/EchoMissionResponse.cs:84]: I saw it was null one time from Echo, and we have a check:
`if (planItem.PoseId is null)` after so that it throws a meaningfull error message instead of crashing when trying to load the json from Echo

File: backend/api/Services/MissionLoaders/MissionLoaderInterface.cs (hunk 4 above)
Comment (Christdej) [reply backend/api/Services/MissionLoaders/MissionLoaderInterface.cs:13]: Probably not, ill remove it =)

File: backend/api.test/Database/Models.cs (hunk 7 above)
Comment (Christdej) [reply backend/api.test/Database/Models.cs:76]: This tests only tests function that are in the test folder and not used in the rest of the application, this should probably have been a separate commit indeed explaining why they are deleted

File: backend/api/Configurations/CustomServiceConfigurations.cs (hunk 2 above)
Comment (Christdej) [reply backend/api/Configurations/CustomServiceConfigurations.cs]: Removed

**Review by andchiind — APPROVED** [review]
